tests/services/orders/submitter/test_order_submitter.py

```diff
@@ -200,16 +206,117 @@ def test_get_ticket_tags(
 ):
     """Test that the correct tags are generated based on the order and order type."""
 
-    # GIVEN an order with existing data
+    # GIVEN an order with existing data and no external samples
     order: OrderWithCases = request.getfixturevalue(order_fixture)
 
+    store: Store = create_autospec(Store)
+    store.get_application_by_tag = lambda tag: Application(is_external=False)
+
     # WHEN getting the ticket tags
-    tags = get_ticket_tags(order=order, order_type=order_type)
+    tags: list[str] = get_ticket_tags(order=order, order_type=order_type, status_db=store)
 
     # THEN the tags should be correct
     assert tags == expected_tags
 
 
+def test_get_ticket_tags_with_external_data_sample_for_order_with_samples():
+
+    # GIVEN an order with two samples, one including external data
+    store: Store = create_autospec(Store)
+    store.get_application_by_tag = lambda tag: (
+        Application(is_external=True) if tag == "external_app" else None
+    )
+
+    order = OrderWithSamples(
+        customer="test_customer",
+        delivery_type=DataDelivery.ANALYSIS_FILES,
+        name="order with external data sample",
+        project_type=OrderType.MIP_DNA,
+        samples=[
+            ValidationSample(
+                application="missing_app",
+                container=ContainerEnum.no_container,
+                name="ExternalSample",
+            ),
+            ValidationSample(
+                application="external_app",
+                container=ContainerEnum.no_container,
+                name="ExternalSample",
+            ),
+        ],
+    )
+
+    # WHEN getting the ticket tags
+    tags: list[str] = get_ticket_tags(order=order, order_type=order.order_type, status_db=store)
+
+    # THEN the tags should include external-data
+    assert "external-data" in tags
+
+
+def test_get_ticket_tags_with_external_data_for_order_with_new_case():
+
+    # GIVEN an order with a new case and a new sample with external data
+    store: Store = create_autospec(Store)
+    store.get_application_by_tag = lambda tag: (
+        Application(is_external=True) if tag == "external_app" else None
+    )
+
+    order = OrderWithCases(
+        delivery_type=DataDelivery.BAM,
+        cases=[
+            ValidationCase(
+                name="new-case",
+                samples=[
+                    BalsamicSample(
```

Comment:
not relevant for the test but a Balsamic sample would not have BAM as delivery type

Reply:
discussed in person